### ISSUE TITLE
Default values for optional  positional arguments

### ISSUE THREAD

**Issue by Paddy3118** [docopt/docopt#214]
In the following:

<pre>
Usage:
  block_lines (rep|exp) [options]  [INPUT-FILE [OUTPUT-FILE]]
  block_lines gen [options]  [OUTPUT-FILE]
  block_lines (-h|--help|--version)

Modes:
  rep               Find the repeated blocks of lines.
  exp               Expand the output from rep.
  gen               (Random gen of output suitable for example rep mode processing).

Options:
  -h --help         Show this screen.
  -v --version      Show version.
  -w N, --width N   Digits in field for repetition count [default: 4].
  -d --debug        Debug:
                      * In rep mode remove anything after '#' on all input lines.
                      * In gen mode annotate with repeat counts after '#'.

I/O Defaults:
  INPUT-FILE        Defaults to stdin.
  OUTPUT-FILE       Defaults to stdout.
</pre>

I would like to specify default values for INPUT-FILE and OUTPUT-FILE but instead have to look for None values and replace them with defaults manually, 

Is there a better way?

**Comment by azag0** [comment]
I personally try to stick to the design rule that if an optional positional argument has a default textual value, it should be an option. In your case, there is actually none, the behaviour is different, say

```python
if args['INPUT-FILE']:
    s = open(args['INPUT-FILE'])
else:
    s = sys.stdin
```

**Comment by poleguy** [comment]
I agree with Paddy3118. It should be added, as it's not that strange a request. Docopt shouldn't force you to work around this deficiency.

**Comment by dradetsky** [comment]
This seems to have been around for some time, but I'd like it as well. I came here after trying the same default syntax as for options, and being surprised it didn't work. Surely it can be made to work.

**Comment by yunake** [comment]
wow, really surprised this doesn't work with docopt. this is such a common thing amongst standard unix tools. this particular stdin/stdout example is a corner-case, and @azag0 argument against it is valid, but the default values for positional parameters do not end with files do they

**Comment by keleshev** [comment]
This is how to accomplish this:

```python
from docopt import docopt
args = docopt("usage: prog <input-file>")
file = open(args["<input-file>"] or "default-file.txt")
```

I would be interested to hear in what ways the above would not be adequate.

**Comment by yunake** [comment]
this is perfectly valid option @keleshev, thank you. this is what i ended up doing too. do you think it's worth documenting this somewhere outside of this bugreport?

**Comment by Paddy3118** [comment]
@keleshev, unfortunately your fix does not work in the docopt manner.  It jars. 
A docopt solution would involve reading the default from the help text,  ideally

**Comment by GhostofGoes** [comment]
So the solution is to encode the defaults wherever in your code you're parsing the arguments. Isn't that somewhat contrary to the purpose of docopt?

**Comment by scharch** [comment]
Here's another vote for docopt handling this natively

**Comment by alexf101** [comment]
I've found this particularly awkward in the Ruby code:

`value = options.fetch('--thing', 'DEFAULT')`

Naively, you'd expect this to work, but you'll get `value = nil`, since docopt sets the parameter to nil instead of leaving it unspecified.

If docopt actually provided a syntax for specifying default values, we'd avoid odd little bugs like this.
